### DISCUSSION TITLE
Breakpoints within a Task

### DISCUSSION THREAD

**Discussion by rrirower** [rokucommunity/brighterscript#895] [Q&A]
Is there any way to set a breakpoint within a Task?  I'm trying to debug some code within a Task object.  I can set a breakpoint, but, when it is hit, I can't examine any variables because the debugger seems to be hung (ie. the Locals window shows a spinning circle).

**Comment by TwitchBronBron** [comment]
I'm assuming you're using the [BrightScript Language](https://marketplace.visualstudio.com/items?itemName=RokuCommunity.brightscript) extension for VSCode? If so, yes, you can definitely set breakpoints within tasks. There are some issues with the telnet debugger right now, it seems to get a bit broken occasionally. You can try setting `"enableDebugProtocol": true` in your launch.json, that provides a much more stable debug experience (for most people).

**Comment by rrirower** [comment]
Thanks.  I'll give that a try.
